backend/main/maps/views.py

```diff
@@ -155,3 +157,116 @@ def get_all_icons(request, route_id):
         i += 1
 
     return Response(data)
+
+from math import sqrt
+def calculate_distance(sx, sy, ex, ey):
+    return sqrt((sx - ex)**2 + (sy - ey)**2)
+
+import json
+@api_view(['GET'])
+def get_summary(request):
+    userid = request.user.id
+    route = Route.objects.filter(user=request.user.id).last()
+    # print(route.route_name)
+    datalist = []
+
+    #TODO: calculate total distance for the first time
+    total_distance = route.total_distance
+    if total_distance == None:
+        coords = route.coordinates.all()
+        if len(coords) == 1:
+            pass #or whatever breaks me out of the outer if statement
+        
+        total_distance = 0.0
+
+        s_lat = coords[0].latitude
+        s_lon = coords[0].longitude
+
+        for i, coord in enumerate(coords):
+            if i == 0:
+                pass
+            
+            e_lat = coords[i].latitude
+            e_lon = coords[i].longitude
+
+            total_distance += calculate_distance(s_lat, s_lon, e_lat, e_lon)
+            s_lat, s_lon = e_lat, e_lon
+        
+        route.total_distance = total_distance
+        route.save(update_fields=['total_distance'])
+    datalist.append(total_distance)
+
+    total_time = route.end_time - route.start_time #datetime.timedelta
+    datalist.append(str(total_time))
+
+    total_hours = total_time.total_seconds() / 3600.0
+    avg_speed = total_distance / total_hours
+    datalist.append(avg_speed)
+
+    pee_stops = len(route.peeIcon.all())
+    poop_stops = len(route.poopIcon.all())
+    water_breaks = len(route.drinkIcon.all())
+    datalist.append(pee_stops)
+    datalist.append(poop_stops)
+    datalist.append(water_breaks)
+
+    #TODO: serialize into JSON object but idk if this is the right way to do it
+    titles = ['Distance', 'Time', 'Avg Speed (mph)', 'Pee Stops', 'Poop Drops', 'Water Breaks']
```

Comment:
It's better to use a serializer because it handles the conversion to json and thus is easier to read.